sycl/doc/extensions/experimental/sycl_ext_oneapi_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,644 @@
+
+## Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification section 6.3.3 "Feature test macros".  Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_MATRIX` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's APIs the implementation supports.
+
+[frame="none",options="header"]
+|======================
+|Value |Description
+|1     |Initial extension JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported 
+|2     |joint matrix type has a new `use` parameter. `layout` on matrix is unused. JIT implementation on Intel AMX and DPAS. load, store, mad, fill, piece-wise operations, and the query interface are supported 
+|======================
+
+## New `joint_matrix` class
+We introduce a new class called `joint_matrix`. The user needs to specify the type of the elements, shape, the matrix use, the memory layout, and the memory scope of the matrix. This results in the following description:
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+template <typename T, use Use,
+          size_t Rows=sycl::dynamic_extent, size_t Cols=sycl::dynamic_extent,
+          layout Layout = layout::unused, typename Group = sub_group>
+struct joint_matrix {
+    joint_matrix(Group g) {}
+};
+}
+```
+
+#### Shape
+The same class, `joint_matrix`, should handle both cases where sizes are constant (GPU case) and when sizes are variable (CPU case). Note that a Intel AMX 2d tile register permits sizes up to 1024 (16rowsx64cols) bytes. The ability to define only one interface for both makes it possible to give the user a way to make use of the flexibility introduced by the CPU but at the same time save resources on the GPU. We use `sycl::dynamic_extent`  to differentiate between static and dynamic sizes.
+
+IMPORTANT: In the current implementation, only the static extent is supported
+
+#### Use
+Specifying the usage of the matrix: matrix left (A), matrix right (B) or accumulator +(C)+ is required by backend implementations to reason about the layout of the matrix in registers.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+enum class use {
+  a,
+  b,
+  accumulator
+};
+}
+```
+
+IMPORTANT: In both AMX and DPAS support, the `use` template parameter is required
+
+
+#### Layout
+Besides row major and column major layouts, `layout` is flexible enough to introduce custom layouts such as packed layout.
+
+```c++
+namespace sycl::ext::oneapi::experimental::matrix {
+enum class layout {
+  row_major,
+  col_major,
+  packed,
+  unused
+};
+}
+```
+
+IMPORTANT: In both AMX and DPAS support, the `layout` template parameter is unused and will be ignored if specified
```

Comment:
This does not match the code synopses you have below, which require they `layout` template parameter to be `layout::unused`.

Reply:
are you referring to this code?
```c++
joint_matrix<int8_t, use::a, tM, tN> tA(sg);
```
yes this code is considering layout to be "unused".
What this statement is saying is that:  in joint matrix, layout is optional. By default is it "unused". In both AMX and DPAS support, layout is unused. If it is specified for tensorcores support (where it is required. Jack will add documentation on top of this once this PR is merged), it is ignored in AMX and DPAS implementations. 
Does this make sense?

Reply:
I'm referring to code synopsis like this:

```
template <typename Group, typename T, size_t NumRows, size_t NumCols,
          use Use, access::address_space Space>
void joint_matrix_load(Group sg,
  joint_matrix<T, Use, NumRows, NumCols, layout::unused, Group> &res,
  multi_ptr<T, Space> src, size_t stride, layout memL);
```

That synopsis says that the matrix parameter must be `layout::unused`.  Any other layout value will cause a compilation error.  Therefore, the value is not ignored.  The same situation exists for the synopses of `joint_matrix_store` and `joint_matrix_mad`.

Reply:
You are right, I am adding Layout on these three functions as layout can have a different value for the CUDA backend.